tests/core/test_jose/test_jwe.py

```diff
@@ -166,78 +169,656 @@ def test_ase_gcm_jwe_invalid_key(self):
+
+        dk_at_bob = alg.deliver_at_recipient(
+            bob_static_key, alice_static_pubkey, alice_ephemeral_pubkey, headers, 256, None)
+        self.assertEqual(dk_at_bob, dk_at_alice)
+
+    def test_ecdh_1pu_key_agreement_computation_appx_b(self):
+        # https://datatracker.ietf.org/doc/html/draft-madden-jose-ecdh-1pu-04#appendix-B
+        alice_static_key = {
+            "kty": "OKP",
+            "crv": "X25519",
+            "x": "Knbm_BcdQr7WIoz-uqit9M0wbcfEr6y-9UfIZ8QnBD4",
+            "d": "i9KuFhSzEBsiv3PKVL5115OCdsqQai5nj_Flzfkw5jU"
+        }
+        bob_static_key = {
+            "kty": "OKP",
+            "crv": "X25519",
+            "x": "BT7aR0ItXfeDAldeeOlXL_wXqp-j5FltT0vRSG16kRw",
+            "d": "1gDirl_r_Y3-qUa3WXHgEXrrEHngWThU3c9zj9A2uBg"
+        }
+        charlie_static_key = {
+            "kty": "OKP",
+            "crv": "X25519",
+            "x": "q-LsvU772uV_2sPJhfAIq-3vnKNVefNoIlvyvg1hrnE",
+            "d": "Jcv8gklhMjC0b-lsk5onBbppWAx5ncNtbM63Jr9xBQE"
+        }
+        alice_ephemeral_key = {
+            "kty": "OKP",
+            "crv": "X25519",
+            "x": "k9of_cpAajy0poW5gaixXGs9nHkwg1AFqUAFa39dyBc",
+            "d": "x8EVZH4Fwk673_mUujnliJoSrLz0zYzzCWp5GUX2fc8"
+        }
+
+        headers = OrderedDict({
+            "alg": "ECDH-1PU+A128KW",
+            "enc": "A256CBC-HS512",
+            "apu": "QWxpY2U",
+            "apv": "Qm9iIGFuZCBDaGFybGll",
+            "epk": OrderedDict({
+                "kty": "OKP",
+                "crv": "X25519",
+                "x": "k9of_cpAajy0poW5gaixXGs9nHkwg1AFqUAFa39dyBc"
+            })
+        })
+
+        cek = b'\xff\xfe\xfd\xfc\xfb\xfa\xf9\xf8\xf7\xf6\xf5\xf4\xf3\xf2\xf1\xf0' \
+              b'\xef\xee\xed\xec\xeb\xea\xe9\xe8\xe7\xe6\xe5\xe4\xe3\xe2\xe1\xe0' \
+              b'\xdf\xde\xdd\xdc\xdb\xda\xd9\xd8\xd7\xd6\xd5\xd4\xd3\xd2\xd1\xd0' \
+              b'\xcf\xce\xcd\xcc\xcb\xca\xc9\xc8\xc7\xc6\xc5\xc4\xc3\xc2\xc1\xc0'
+
+        iv = b'\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f'
+
+        payload = b'Three is a magic number.'
+
+        alg = JsonWebEncryption.ALG_REGISTRY['ECDH-1PU+A128KW']
+        enc = JsonWebEncryption.ENC_REGISTRY['A256CBC-HS512']
+
+        alice_static_key = OKPKey.import_key(alice_static_key)
+        bob_static_key = OKPKey.import_key(bob_static_key)
+        charlie_static_key = OKPKey.import_key(charlie_static_key)
+        alice_ephemeral_key = OKPKey.import_key(alice_ephemeral_key)
+
+        alice_static_pubkey = alice_static_key.get_op_key('wrapKey')
+        bob_static_pubkey = bob_static_key.get_op_key('wrapKey')
+        charlie_static_pubkey = charlie_static_key.get_op_key('wrapKey')
+        alice_ephemeral_pubkey = alice_ephemeral_key.get_op_key('wrapKey')
+
+        protected_segment = json_b64encode(headers)
+        aad = to_bytes(protected_segment, 'ascii')
+
+        ciphertext, tag = enc.encrypt(payload, aad, iv, cek)
```

Comment:
It would be great to check that Ze, Zs, Z (and all other parameters that are not checked) are equal to test vector values from Appendix B.